Bottom: 511cb83f879e9c59b5597df30856d4a72844a8cc
Top:    01e0c526c77aef41cdc2ba9aedfd32930c02e6ee
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-17 02:59:33 +0100

Refresh of add-support-for-depth-to-trian

---

diff --git a/src/hid/common/hidgl.h b/src/hid/common/hidgl.h
index 2ce5283..c6f79c4 100644
--- a/src/hid/common/hidgl.h
+++ b/src/hid/common/hidgl.h
@@ -43,16 +43,9 @@ hidgl_add_triangle (triangle_buffer *buffer,
                     GLfloat x2, GLfloat y2,
                     GLfloat x3, GLfloat y3)
 {
-  buffer->triangle_array [buffer->coord_comp_count++] = x1;
-  buffer->triangle_array [buffer->coord_comp_count++] = y1;
-  buffer->triangle_array [buffer->coord_comp_count++] = global_depth;
-  buffer->triangle_array [buffer->coord_comp_count++] = x2;
-  buffer->triangle_array [buffer->coord_comp_count++] = y2;
-  buffer->triangle_array [buffer->coord_comp_count++] = global_depth;
-  buffer->triangle_array [buffer->coord_comp_count++] = x3;
-  buffer->triangle_array [buffer->coord_comp_count++] = y3;
-  buffer->triangle_array [buffer->coord_comp_count++] = global_depth;
-  buffer->triangle_count++;
+  hidgl_add_triangle_3D (x1, y1, global_depth,
+                         x2, y2, gloabl_depth,
+                         x3, y3, global_depth);
 }
 
 static inline void
@@ -61,10 +54,6 @@ hidgl_add_triangle_3D (triangle_buffer *buffer,
                        GLfloat x2, GLfloat y2, GLfloat z2,
                        GLfloat x3, GLfloat y3, GLfloat z3)
 {
-  if (buffer->triangle_count == TRIANGLE_ARRAY_SIZE) {
-    fprintf (stderr, "hidgl_add_triangle_3D: Attempt to add past end of array. Will now segfault.\n");
-    *(char *)0 = 0;
-  }
   buffer->triangle_array [buffer->coord_comp_count++] = x1;
   buffer->triangle_array [buffer->coord_comp_count++] = y1;
   buffer->triangle_array [buffer->coord_comp_count++] = z1;
